subsys/bluetooth/host/id.c

```diff
@@ -361,14 +363,19 @@ int bt_id_set_adv_private_addr(struct bt_le_ext_adv *adv)
 		return 0;
 	}
 
-	err = bt_rpa_create(bt_dev.irk[adv->id], &rpa);
-	if (!err) {
-		err = bt_id_set_adv_random_addr(adv, &rpa);
-		if (!err) {
-			atomic_set_bit(adv->flags, BT_ADV_RPA_VALID);
+	if (bt_addr_le_eq(&bt_dev.rpa[adv->id], BT_ADDR_LE_NONE)) {
+		err = bt_rpa_create(bt_dev.irk[adv->id], &bt_dev.rpa[adv->id].a);
+		if (err) {
+			return err;
 		}
 	}
 
+	err = bt_id_set_adv_random_addr(adv, &bt_dev.rpa[adv->id].a);
+
```

Comment:
minor: remove this line (just to be consistent with the style - no empty line between assignment to variable and testing its value)

Reply:
Done